generator/file_reader.go

```diff
@@ -0,0 +1,95 @@
+package generator
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const (
+	MaxHeaderBytes = 50 * 1024 // 50kb for the header should be enough?
```

Comment:
This seems like an unnecessary optimization. At a minimum, I would want to make it so large as to be somewhat ridiculous.